pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -754,7 +754,7 @@ func (pl *dynamicResources) Filter(ctx context.Context, cs *framework.CycleState
 			// would just get allocated again for a random node,
 			// which is unlikely to help the pod.
 			if claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer {
-				state.unavailableClaims.Insert(unavailableClaims...)
+				state.unavailableClaims.Insert(index)
```

Comment:
Would it make sense to cover this with a unit or e2e tests?

Reply:
There is a test which exercises deallocation, just not in a way that mattered here.

As this is such a subtle difference, I am not sure how to write a test that covers this.

Reply:
One test also makes sense to me.
/approve